cmd/snap/cmd_run.go

```diff
@@ -216,6 +220,68 @@ func snapRunHook(snapName, snapRevision, hookName string) error {
 	return runSnapConfine(info, hook.SecurityTag(), snapName, "", hook.Name, nil)
 }
 
+func migrateXauthority(info *snap.Info) error {
+	u, err := userCurrent()
+	if err != nil {
+		return fmt.Errorf(i18n.G("cannot get the current user: %v"), err)
+	}
+
+	xauthPath := osGetenv("XAUTHORITY")
+	if len(xauthPath) == 0 || !osutil.FileExists(xauthPath) {
+		// Nothing to do for us. Most likely running outside of any
+		// graphical X11 session.
+		return nil
+	}
```

Comment:
@jdstrand I've implemented this now. Can you check if it matches now your expectations? The code is still a bit rough and tests are not adjusted but wanted to have these changes up as quick as possible to not loose time on your general feedback. Will work on optimizations and test cases next.